src/triage/component/architect/feature_dictionary.py

```diff
@@ -0,0 +1,64 @@
+from triage.component.architect.utils import remove_schema_from_table_name
+from triage.util.structs import FeatureNameList
+from collections.abc import Iterable
+from collections import MutableMapping
+
+
+class FeatureDictionary(MutableMapping):
+    """A feature dictionary, consisting of table names as keys and column names as values
+
+    If a list of feature_blocks is passed, will initialize the feature dictionary with their data.
+    """
+    def __init__(self, feature_blocks=None, *args, **kwargs):
+        self.tables = dict()
+        self.update(dict(*args, **kwargs))  # use the free update to set keys
```

Comment:
I believe `update` and `dict` have the same interface; I don't think it's necessary to pass these thru `dict`.